app/assets/javascripts/components/high_order/editable.cjsx

```diff
@@ -28,13 +28,14 @@ Editable = (Component, Stores, Save, GetState, Label) ->
       if permissions && @state.editable
         unless save_only
           className = 'controls'
-          cancel = (
-            <button onClick={@cancelChanges} className='button'>{I18n.t('editable.cancel')}</button>
-          )
+          unless SaveOnly
```

Comment:
Yeah, good point. This was the hackiest bit that I did. The 'Editable' component takes a provided set of controls, and mixes in another one to toggle between Edit mode and adds a Save and Cancel button when edit mode is activated.

The students_list relies on this to properly render the *other* controls for Enrollment and Notify Students, but I wanted to change the assignments part to be just a toggle that adds or removes the Assign buttons from each cell. The original save_only option I couldn't figure out how to utilize cleanly, so I added that other one.

Okay, yeah. I knew this was a bad thing when I did it, but I figured... the Editable component is a complex mess anyway, so why not pile on?

Sigh.